tests/snuba/api/endpoints/test_organization_events_stats.py

```diff
@@ -571,3 +572,240 @@ def test_invalid_interval(self):
                 },
             )
         assert response.status_code == 400
+
+
+class OrganizationEventsStatsTopNEvents(APITestCase, SnubaTestCase):
+    def setUp(self):
+        super(OrganizationEventsStatsTopNEvents, self).setUp()
+        self.login_as(user=self.user)
+
+        self.day_ago = before_now(days=1).replace(hour=10, minute=0, second=0, microsecond=0)
+
+        self.project = self.create_project()
+        self.project2 = self.create_project()
+        self.user2 = self.create_user()
+        transaction_data = load_data("transaction")
+        transaction_data["start_timestamp"] = iso_format(self.day_ago + timedelta(minutes=2))
+        transaction_data["timestamp"] = iso_format(self.day_ago + timedelta(minutes=4))
+        for i in range(3):
+            self.event1 = self.store_event(
+                data={
+                    "event_id": "a{}".format(i) * 16,
+                    "message": "poof",
+                    "timestamp": iso_format(self.day_ago + timedelta(minutes=2)),
+                    "user": {"email": self.user.email},
+                    "fingerprint": ["group3"],
+                },
+                project_id=self.project.id,
+            )
+            self.event2 = self.store_event(
+                data={
+                    "event_id": "b{}".format(i) * 16,
+                    "message": "voof",
+                    "timestamp": iso_format(self.day_ago + timedelta(hours=1, minutes=2)),
+                    "fingerprint": ["group2"],
+                    "user": {"email": self.user2.email},
+                },
+                project_id=self.project2.id,
+            )
+            self.store_event(
+                data={
+                    "event_id": "c{}".format(i) * 16,
+                    "message": "very bad",
+                    "timestamp": iso_format(self.day_ago + timedelta(minutes=2)),
+                    "fingerprint": ["group1"],
+                    "user": {"email": "foo@example.com"},
+                },
+                project_id=self.project.id,
+            )
+            self.event3 = self.store_event(
+                data={
+                    "event_id": "d{}".format(i) * 16,
+                    "message": "very bad",
+                    "timestamp": iso_format(self.day_ago + timedelta(hours=1, minutes=2)),
+                    "fingerprint": ["group1"],
+                    "user": {"email": "foo@example.com"},
+                },
+                project_id=self.project.id,
+            )
+            self.transaction = self.store_event(transaction_data, project_id=self.project.id)
```

Comment:
Why do you do this 3 times? Shouldn't you at least overwrite some part of the data so you can tell the transactions apart?

Reply:
(note: resolved other convos cause they were with reference events and i removed those changes, Replying to this one cause tests are still using this pattern)
I just wanted to create some counts that weren't 1 everywhere, is that problematic? I could offset each event so that they differ.

Reply:
It's just that I don't think these will be recorded as 3 separate transactions. At least I've had that happen to me in the tests if I didn't change some of the data between events.

Reply:
So they become separate as long as the event_id is different, I now create a unique one for each item in the test using their indexes in the list:
https://github.com/getsentry/sentry/pull/17987/files#diff-3ab46b8a04b6d001891800a63665c90bR657-R662